App.tsx

```diff
@@ -54,18 +59,89 @@ function ScreenNavigators() {
 }
 
 const MyComponent = () => {
+  const dispatch = useDispatch();
   const state = useSelector(state => state.login);
+  const loadingStatus = useRef('stoped');
   const {loading} = state;
+  const [networkStatus, setNetworkStatus] = useState(true);
+
+  const indicatorTop = useRef(new Animated.Value(0)).current;
+
+  /**
+   * judge login status
+   */
+  useEffect(() => {
+    dispatch(getLoginInfo());
+  });
+
+  useEffect(() => {
+    if (loading && loadingStatus.current === 'stoped') {
+      loadingStatus.current = 'started';
+      Animated.timing(indicatorTop, {
+        toValue: 50,
+        duration: 1000,
+      }).start();
+    } else {
+      loadingStatus.current = 'stoped';
+      indicatorTop.setValue(0);
+    }
+  }, [indicatorTop, loading]);
+
+  const tipsCallback = useCallback(() => {
+    setNetworkStatus(true);
+  }, [setNetworkStatus]);
+
+  /**
+   * testing network connection
+   *
+   */
+
+  useEffect(() => {
+    NetInfo.fetch().then((state: object) => {
+      if (!state.isConnected) {
+        setNetworkStatus(false);
+      }
+    });
+    const unsubscribe = NetInfo.addEventListener((state: NetInfoState) => {
+      console.log('changed');
+      if (!state.isConnected) {
+        setNetworkStatus(false);
+      }
+    });
+
+    return unsubscribe;
+  }, []);
```

Comment:
Please never cheat deps! It may lead to unexpected errors
[Read here](https://overreacted.io/a-complete-guide-to-useeffect/) please why it is important